src/components/widgets/support-case-widget.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Table,
+  TableVariant,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+import { Button } from '@patternfly/react-core/dist/dynamic/components/Button';
+import { EmptyState } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateBody } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateIcon } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { EmptyStateVariant } from '@patternfly/react-core/dist/dynamic/components/EmptyState';
+import { Stack } from '@patternfly/react-core/dist/dynamic/layouts/Stack';
+import { StackItem } from '@patternfly/react-core/dist/dynamic/layouts/Stack';
+import { Title } from '@patternfly/react-core/dist/dynamic/components/Title';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/dynamic/icons/external-link-alt-icon';
+import { Label } from '@patternfly/react-core/dist/dynamic/components/Label';
+import HeadsetIcon from '@patternfly/react-icons/dist/dynamic/icons/headset-icon';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
+import SkeletonTable from '@patternfly/react-component-groups/dist/dynamic/SkeletonTable';
+
+export type Case = {
+  id: string;
+  caseNumber: string;
+  summary: string;
+  lastModifiedById: string;
+  severity: string;
+  status: string;
+};
+
+const SupportCaseWidget: React.FunctionComponent = () => {
+  const [cases, setCases] = useState<Case[]>([]);
+  const MAX_ROWS = 5;
+  const chrome = useChrome();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const columnNames = {
+    caseId: 'Case ID',
+    issueSummary: 'Issue summary',
+    modifiedBy: 'Modified by',
+    severity: 'Severity',
+    status: 'Status',
+  };
+
+  const fetchSupportCases = async () => {
```

Comment:
How about we move this function away from component and accept `chrome` argument? The data processing can then happen in the `then` clause
```JS
  const fetchSupportCases = async (chrome) => {
    const token = await chrome.auth.getToken();
    const user = await chrome.auth.getUser();
    const url = 'https://api.access.redhat.com/support/v1/cases/filter';
    const options = {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${token}`,
        'Content-Type': 'application/json',
        Accept: 'application/json',
      },
      body: JSON.stringify({
        createdBySSOName: `${user?.identity.user?.username}`,
      }),
    };

    try {
      const response = await fetch(url, options);
      return await response.json();
    } catch (error) {
      console.error('Unable to fetch support cases', error);
    }
  };

  useEffect(() => {
    setIsLoading(true);
    fetchSupportCases(chrome).then((cases) => {
        if (cases) {
          setCases(cases);
        } else {
            // some kind of error
        }
        setIsLoading(false);
    });
  }, []);
```